tests/test_profile_data.py

```diff
@@ -12,17 +13,21 @@
 )
 from splink.spark.linker import SparkLinker
 from splink.sqlite.linker import SQLiteLinker
+from tests.decorator import mark_with_dialects_excluding
 
 from .basic_settings import get_settings_dict
 
 
-def generate_raw_profile_dataset(columns_to_profile, linker):
-    linker._initialise_df_concat()
+def generate_raw_profile_arrays_dataset(columns_to_profile, linker, cast_arrays_as_str):
+
+    df_concat = linker._initialise_df_concat()
+
+    array_cols = df_concat.get_array_cols()
```

Comment:
This currently warns the user that `Profiling arrays is not implemented for this linker` - you probably want to hide this away behind a check to see if the user wants to check for array cols.

Reply:
Do you have an example of what you mean by hiding behind a check?

Reply:
Essentially...
`array_cols = [] if cast_arrays_as_str else df_concat.get_array_cols()`

Otherwise you're going to warn the user that we can't unpack arrays for all profiling jobs, even if they don't request that behaviour.